pkg/utils/kube/kube.go

```diff
@@ -367,31 +368,20 @@ func GetDeploymentReplicas(u *unstructured.Unstructured) *int64 {
 	return &val
 }
 
-// RetryUntilSucceed keep retrying given action with specified timeout until action succeed or specified context is done.
-func RetryUntilSucceed(action func() error, desc string, ctx context.Context, timeout time.Duration) {
-	ctxCompleted := false
-	stop := make(chan bool)
-	defer close(stop)
-	go func() {
-		select {
-		case <-ctx.Done():
-			ctxCompleted = true
-		case <-stop:
-		}
-	}()
-	for {
+// RetryUntilSucceed keep retrying given action with specified interval until action succeed or specified context is done.
+func RetryUntilSucceed(action func() error, desc string, ctx context.Context, interval time.Duration) {
```

Comment:
It might be cleaner to have a signature like this:
```go
func RetryUntilSucceed(ctx context.Context, interval time.Duration, desc string, action func() error)
```
context should go first by convention and it's always nicer to pass functions as the last argument because then if it's a closure, it doesn't have more arguments after it, which reads much better.

But I wanted to just fix the bug and not touch the signature here. If you'd like, I can send a follow up PR.

Reply:
Agree, the proposed signature is cleaner and easier to use. This function is used only once in GitOps Engine and twice in Argo CD repo, so it will be easy to migrate to a new signature.